### DISCUSSION TITLE
KEDA eventhub scaler not using FQDN for network access?

### DISCUSSION THREAD

**Discussion by HenrikDK** [kedacore/keda#3772] [Q&A / Need Help] [answered]
Our IT department has a requirement to use the Fully Qualified Domain Name (FQDN) for any access that goes through the firewall, currently KEDA seems to use IP to access the eventhub:

`
1.6666068669775739e+09	ERROR	azure_eventhub_scaler	unable to get runtimeInfo for isActive	{"error": "dial tcp 10.237.198.164:5671: connect: connection timed out"}
github.com/kedacore/keda/v2/pkg/scalers.(*azureEventHubScaler).IsActive
	/workspace/pkg/scalers/azure_eventhub_scaler.go:254
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).IsScaledObjectActive
	/workspace/pkg/scaling/cache/scalers_cache.go:93
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:278
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:149
`

Is there anyway to change this behaviour?

**Comment by JorTurFer** [comment, marked as answer]
We use the SDK to access the EventHub, and we pass the domain to it. The shown error is an internal error from the SDK, but it's something internal of it, KEDA doesn't control that. But I don't get the problem, you are using the FQDN and that is resolved into an IP address that it's used, there is not a way to use the FQDN all the time, that's like a domain, it's resolved to an IP address and based on your logs, I'd say that it works because I see a private IP address

**Reply by HenrikDK** [reply]
Hi 

Thanks for the refresh on TCP IP networking :) don't know why I didn't remember that when i wrote the question :D. You are of course correct it seems that we are using Azure firewalls FQDN filtering rules that rely on caching dns entries.

https://learn.microsoft.com/en-us/azure/firewall/fqdn-filtering-network-rules

My guess is that something is up with our internal dns resolution and for some reason the IP KEDA is "dialing" isn't being matched by the FQDN in the azure firewall, I will bring this issue up with our IT department.
